Emby.Drawing/ImageProcessor.cs

```diff
@@ -246,48 +261,111 @@ private ImageFormat GetOutputFormat(IReadOnlyCollection<ImageFormat> clientSuppo
         /// <summary>
         /// Gets the cache file path based on a set of parameters.
         /// </summary>
-        private string GetCacheFilePath(string originalPath, ImageDimensions outputSize, int quality, DateTime dateModified, ImageFormat format, bool addPlayedIndicator, double percentPlayed, int? unwatchedCount, int? blur, string backgroundColor, string foregroundLayer)
+        private string GetCacheFilePath(
+            string originalPath,
+            int? width,
+            int? height,
+            int? maxWidth,
+            int? maxHeight,
+            int? fillWidth,
+            int? fillHeight,
+            int quality,
+            DateTime dateModified,
+            ImageFormat format,
+            bool addPlayedIndicator,
+            double percentPlayed,
+            int? unwatchedCount,
+            int? blur,
+            string backgroundColor,
+            string foregroundLayer)
         {
-            var filename = originalPath
-                + "width=" + outputSize.Width
-                + "height=" + outputSize.Height
-                + "quality=" + quality
-                + "datemodified=" + dateModified.Ticks
-                + "f=" + format;
+            System.Text.StringBuilder filename = new System.Text.StringBuilder(128);
```

Comment:
```suggestion
            var filename = new StringBuilder(128);
```

Reply:
Have to import `System.Text` though.

Reply:
I wasn't quite sure I wanted to import `System.Text` just for `StringBuilder`. Should I always do that?

Reply:
There's no functional difference

Reply:
Well, the functional difference would be bringing all of `System.Text.*` into scope, no?

Reply:
I suppose, but the IL remains the same